server/utils/pagination/pagination.ts

```diff
@@ -0,0 +1,95 @@
+import { Page } from '../../models/pagination'
+
+interface MojPaginationArgs {
+  classes?: string
+  items: (
+    | {
+        type: 'dots'
+      }
+    | {
+        type: 'pageNumber'
+        selected: boolean
+        text: number
+        href: string
+      }
+  )[]
+  previous?: {
+    text: string
+    href: string
+  }
+  next?: {
+    text: string
+    href: string
+  }
+  results?: {
+    from: number
+    to: number
+    count: number
+  }
+}
+
+export default class Pagination {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(private readonly page: Page<any>) {}
+
+  private constructPageItems(pageNumberRange: number[], chosenPageNumber: number): MojPaginationArgs['items'] {
+    return pageNumberRange.map(pageNumber => {
+      return {
+        type: 'pageNumber',
+        selected: chosenPageNumber === pageNumber,
+        text: pageNumber,
+        href: `?page=${pageNumber}`,
+      }
+    })
+  }
+
+  get mojPaginationArgs(): MojPaginationArgs {
+    const { totalPages } = this.page
+    if (totalPages <= 1) {
+      return {
+        items: [],
+      }
+    }
+    const count = this.page.totalElements
+    // page.number is zero indexed
```

Comment:
is this still the case?

Reply:
Unfortunately yes.

You can only make the request non-zero indexed but the response is always zero-indexed.

I assume the reason is to not break the contract of the response.

I think it's still useful to have the request being non-zero indexed because response handling is encapsulated in this single class and all users of pagination won't have to worry about decrementing before sending the request.